tests/test_types.py

```diff
@@ -50,37 +50,74 @@ def test_enum():
 
 
 def test_reference():
-    from alfasim_sdk.types import Reference, TracerType
+    from alfasim_sdk.types import Reference
+    from alfasim_sdk.types import TracerType
+    from alfasim_sdk.models import data_model, container_model
+
+    @data_model(caption='caption')
+    class Data:
+        pass
 
-    class Data1:
+    @container_model(caption='caption', model=Data, icon='')
+    class DataContainer:
+        pass
+
+    class InvalidClass:
         pass
 
     with pytest.raises(TypeError, match="missing 1 required keyword-only argument: 'caption'"):
         Reference(ref_type="")
 
-    with pytest.raises(TypeError, match="arg 1 must be a class"):
+    with pytest.raises(TypeError, match='ref_type must be a class'):
         Reference(ref_type="", caption="caption")
 
-    with pytest.raises(TypeError, match="ref_type must be a valid ALFAsim type"):
-        Reference(ref_type=Data1, caption="caption")
+    with pytest.raises(TypeError, match="ref_type must be an ALFAsim type or a class decorated with 'data_model'"):
+        Reference(ref_type=InvalidClass, caption="caption")
+
+    error_msg = "ref_type must be an ALFAsim type or a class decorated with 'data_model', got a class decorated with 'container_model'"
```

Comment:
If possible, move those checks to a function and reuse them in the MultipleReference test